libraries/chain/controller.cpp

```diff
@@ -491,38 +491,30 @@ struct building_block {
 
       uint32_t get_block_num() const { return block_num; }
 
-      // returns the next proposer schedule version and true if different
+      // returns the next proposer schedule version and true if producers should be proposed in block
       // if producers is not different then returns the current schedule version (or next schedule version)
       std::tuple<uint32_t, bool> get_next_proposer_schedule_version(const vector<producer_authority>& producers) const {
```

Comment:
I don't think this function is needed anymore. 
If `set_proposed_producers` is called in a block, the version should be the version last proposed in a previous block + 1.
The version doesn't have to be stored in the `global_property_object`.  I think it could be tracked in the `block_header_state` just like `finalizer_policy_generation`. Whenever a block has at least one `set_proposed_producers` call, the new `proposer_policy` is included in the `if_extension` with the version being `bb.parent.proposer_policy_generation + 1`

Reply:
We have to store in `global_property_object` in case the transaction is aborted, we need the chainbase rollback to remove the change to the global_property_object. A rollback of a transaction does not rollback any changes in the `block_header_state`.